### ISSUE TITLE
BTDT not remembering some anomalies (randoliths?)

### ISSUE THREAD

**Issue by JonnyOThan** [KSPModStewards/SCANsat#381]
We've noticed an issue on my "twitch plays ksp" stream.  Currently on KSP 1.8.1 using scansat 1.8.14 (according to the version file anyway).

When we use the BTDT near a randolith, it will get marked as visited.  If we switch vessels, scenes, quicksave/load or anything, the anomaly is still marked as visited.  But if we restart KSP, the anomaly is no longer visited.  This seems to only happen to randoliths - other anomalies stay marked across restarts.  This has happened on at least 2 different bodies - eeloo and ceti (from GPP).

We've done a complete orbital anomaly scan, so we have the geocoordinates of each one.  The anomalydetail coverage at the randolith seems to be what is controlling whether the anomaly is marked or not, and after restarting ksp that coordinate is no longer covered.

We can reproduce this quite easily, so I'm willing to test out any ideas you might have.

Interestingly, if we switch to a vessel that is near the randolith, it then gets marked as visited even if there are no active BTDT scanners nearby.

**Comment by DMagic1** [comment]
To be clear, the randoliths are still showing up on the maps, but are marked with grey icons and say "Anomaly: Unknown" when you mouse over it? Or the marker disappears altogether?

I'm not seeing either in my testing, but it's not entirely surprising that something weird could be going on with the BTDT and the random anomalies. I've seen other things like the random anomalies moving to different locations, though.

The root of the problem might be the way that the BTDT scan works. I think I need to just overhaul that altogether to come up with a more robust system.

**Comment by JonnyOThan** [comment]
Right, the location is still known but the AnomalyDetail coverage in that spot is missing.
